src/test/java/com/greenfoxacademy/controllers/ContactControllerTest.java

```diff
@@ -252,6 +252,40 @@ public void badRequestPut() throws Exception {
                 .andExpect(status().isBadRequest());
     }
 
+    @Test
+    public void unauthenticatedContactIdGet() throws Exception {
+        int id = 1;
+        mockMvc.perform(get("/contact/{id}", id)
+                .header("session_token", "abcdefgéééééééééééé"))
+                .andExpect(status().isUnauthorized());
+
+    }
+
+    @Test
+    public void contactIdGet() throws Exception {
+        int id = 1;
+        TestContact testContact = new TestContact("Pista bá", "Fúr, farag", 1L);
+        String validTestJson = createTestJson(testContact);
+
+        mockMvc.perform(post("/contacts")
+                .header("session_token", "abcde")
+                .header("Origin", "https://lasers-cornubite-konnekt.herokuapp.com")
+                .contentType(MediaType.APPLICATION_JSON).content(validTestJson));
+
+        mockMvc.perform(get("/contact/{id}", id)
```

Comment:
As of now, all tests are transactional, meaning, each transaction is rolled back after the completion of a test. So putting the creation of the test contact in a different method is not feasible. Would you consider this 
```
    Contact testcontact = new Contact(userService.findUserById(1L), "John Doe", "FOOBAR");
    contactRepository.save(testcontact);

```
a more appealing solution than the current one?
 ```
       mockMvc.perform(post("/contacts")
                .header("session_token", "abcde")
                .contentType(MediaType.APPLICATION_JSON).content(validTestJson));
```